pkg/util/release/reports.go

```diff
@@ -0,0 +1,63 @@
+package release
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+)
+
+func BuildClusterBreakdownReport(reports []shipper.ClusterCapacityReport) string {
+	res := make([]string, 0, len(reports))
+	for _, r := range reports {
+		breakdown := r.Breakdown
+		if len(breakdown) == 0 {
+			continue
+		}
+		for _, sb := range breakdown {
+			if len(sb.Containers) == 0 {
+				continue
+			}
+			status := sb.Status
+			typ := sb.Type
+			contNames := make([]string, 0, len(sb.Containers))
+			var totalcnt uint32
+			for _, cont := range sb.Containers {
+				var cnt uint32
+				for _, s := range cont.States {
+					cnt += s.Count
+				}
+				contNames = append(contNames, fmt.Sprintf("%s(x%d)", cont.Name, cnt))
+				totalcnt += cnt
+			}
+			res = append(res, fmt.Sprintf(
+				"%d container(s) [%s] are in status: %s:%s",
+				totalcnt,
+				strings.Join(contNames, ","),
+				typ,
+				status))
+		}
+	}
+
+	sort.Strings(res)
+
+	return strings.Join(res, "\n")
+}
+
+func BuildClusterProblemReport(s []shipper.PodStatus) string {
```

Comment:
I worry that problems at a pod level don't tell the entire story. Problems with quotas or a validating webhook denying the creation of pods, for instance, won't show up. I suppose we'll need further changes to the capacity controller to surface that up (presumably by looking at Deployment conditions) before doing changes here though.

Reply:
Thanks for sharing this @juliogreff! I wonder if you see this reasonable to make the release controller to have an application cluster connection pool and communicate to them directly or would you suggest an alternative approach to get deployment condition insights?